### PR TITLE
[APP-41738]pusher ignores Request too large

### PR DESCRIPTION
**Pull request by ericfan17**
test result: 
截圖 2022-06-02 下午6 40 59

https://etcd.io/docs/v3.3/upgrades/upgrade_3_2/#changed-maximum-request-size-limits-3210
(Warning! the format here is slightly off)

https://17media.slack.com/archives/C02Q6EVAWTX/p1652761254698949

https://17media.atlassian.net/browse/APP-41738

### CODE REVIEW

File: pusher/main.go [hunk 1]
```diff
@@ -351,8 +354,14 @@ func Pusher(etcdConn *clientv3.Client, root, etcdRoot string) {
...
 			_, err = etcdConn.Put(ctx, fileP, string(content[k].content))
 		}
-		if err != nil {
-			log.Fatalf("error when setting znode >%s(%s + %s)<. Config server will be inconsistent: %s",
+		// if the key is too large we ignore the error
+		// https://etcd.io/docs/v3.3/upgrades/upgrade_3_2/#changed-maximum-request-size-limits-3210
+		// error msg in the document is slightly off, so we define it here.
+		// ErrRequestTooLarge comes from go.etcd.io/etcd v3.3.27+incompatible, but we are only using string so it should be find
+		if err != nil && (strings.HasPrefix(err.Error(), clientTooLargeErr) || err.Error() == rpctypes.ErrRequestTooLarge.Error()) {
+			fmt.Printf("file %s\n is too large, error: %s\n", fileP, err.Error())
```
Comment (shawnpeng17) [inline pusher/main.go]: Do we need to log it?
If so, we can use  `log.Printf` instead.

File: pusher/main.go [hunk 2]
```diff
@@ -351,8 +354,14 @@ func Pusher(etcdConn *clientv3.Client, root, etcdRoot string) {
 		if !content[k].isDir {
 			_, err = etcdConn.Put(ctx, fileP, string(content[k].content))
 		}
-		if err != nil {
-			log.Fatalf("error when setting znode >%s(%s + %s)<. Config server will be inconsistent: %s",
+		// if the key is too large we ignore the error
+		// https://etcd.io/docs/v3.3/upgrades/upgrade_3_2/#changed-maximum-request-size-limits-3210
+		// error msg in the document is slightly off, so we define it here.
+		// ErrRequestTooLarge comes from go.etcd.io/etcd v3.3.27+incompatible, but we are only using string so it should be find
+		if err != nil && (strings.HasPrefix(err.Error(), clientTooLargeErr) || err.Error() == rpctypes.ErrRequestTooLarge.Error()) {
```
Comment (shawnpeng17) [inline pusher/main.go:361]: `err != nil` can remove

Comment (ericfan17) [reply pusher/main.go:361]: err.Error() will panic
is there a better coding style for this ?

File: pusher/main.go (hunk 1 above)
Comment (ericfan17) [reply pusher/main.go]: updated

File: pusher/main.go (hunk 2 above)
Comment (shawnpeng17) [reply pusher/main.go:361]: Got it.
Or we can use` else` instead of `else if err != nil`

Comment (shawnpeng17) [reply pusher/main.go:361]: Got it

**Review by shawnpeng17 — APPROVED** [review]

**Review by starkhuang17 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -6,6 +6,12 @@ and update etcdclient to v3
 
 deprecates 17media/go-etcd
 
+note that we ignore errors that files are too large to push (limitation for server side: 1.5 MiB, client side: 4 MiB), because they are not etcd meant to maintain
```
Comment (seanhsu17) [inline README.md]: It seems the default value is 2MB for client-side.
```
If not specified, client-side send limit defaults to 2 MiB (1.5 MiB + gRPC overhead bytes) and receive limit to math.MaxInt32. Please see [clientv3 godoc](https://pkg.go.dev/github.com/etcd-io/etcd/clientv3#Config) for more detail.
```
https://github.com/etcd-io/etcd/blob/08407ff7600eb16c4445d5f21c4fafaf19412e24/client/v3/config.go#L46

File: pusher/run.sh [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
 go run main.go \
-  -csi.config.repo.root="$GOPATH/src/github.com/17media/configs/envs/dev"\
-  -csi.config.etcd.root="/configs/envs/dev" \
+  -csi.config.repo.root="$GOPATH/src/github.com/17media/configs/envs/prod"\
+  -csi.config.etcd.root="/configs/envs/prod" \
```
Comment (jerry-yt-chen) [inline pusher/run.sh]: 改這個部分是只會影響local嗎

Comment (ericfan17) [reply pusher/run.sh]: 對應該只影響local沒錯 不過我誤commit了 應該要改回來才對
用prod測的原因是prod的檔案才夠大

**Review by jerry-yt-chen — APPROVED** [review]

**Review by seanhsu17 — APPROVED** [review]

**Review by ivanlo25 — APPROVED** [review]

**Pull request merged**